File: eecs168/2020fall/TR/2020.11.10.classes-private-scope/Account/Account.cpp
//Account.cpp
#include "Account.h"


Account::Account()
{
	//Initialize member variables
	balance = 0;
}

bool Account::deposit(double amount)
{
	if( amount > 0 )
	{	
		balance = balance + amount;
		return (true);
	}
	else
	{
		return (false);
	}	
}

bool Account::withdraw(double amount)
{
	if( amount <= balance && amount > 0 )
	{
		balance = balance - amount;
		return (true);
	}
	else
	{
		return (false);
	}	
}

double Account::checkBalance()
{
	return(balance);
}

File: eecs168/2022summer/2022.07.21.classes-part-3/account.py
#account.py

class Account:
    def __init__(self):
        self._balance = 0.0

    def deposit(self, amount):
        if amount > 0:
            self._balance += amount

    def withdraw(self, amount):
        if amount > 0 and amount <= self._balance:
            self._balance -= amount

    def check_balance(self):
        return self._balance
